int64 time
string[] targets
